### ISSUE TITLE
Add zooming controls for histogram x-axis

### ISSUE THREAD

**Issue by bobular** [VEuPathDB/web-components#70]
There's currently functionality for **range selection** in the histogram and histogram controls

https://veupathdb.github.io/web-components/?path=/story/plots-histogram--range-selection

To add zooming in the x-axis, we could

- add a new control that selects between "select" (default) and "zoom" behaviour.
- when in zoom mode, a "selection" will trigger a new back end request (see API definition for `{date,numeric}-histogram-{bin-width,num-bins}` here: https://veupathdb.github.io/EdaDataService/api.html#resources:/analyses/numeric-histogram-bin-width:post ) with the config.viewportMin and config.viewportMax set.

see also #68 for resetting all the ranges and zooming out.

**Comment by dmfalke** [comment]
I don't agree with toggling the functionality of a range selector as described in the description. I think these should be distinct widgets. @moontrip and I have discussed this and he is incorporating this into #87.

**Comment by bobular** [comment]
Does this mean having just text input-based zooming? (not mouse-based)

**Comment by dmfalke** [comment]
Yes, that was my understanding of what we agreed upon. I could be remembering incorrectly. I think we will want the text inputs regardless, so we can always add a toggle to change the behavior of the mouse gesture in the future.

fwiw, I didn't fully understand that this issue was about the mouse gesture, which may have added to some confusion.

**Comment by bobular** [comment]
Old ticket - but perhaps a "zoom to selected range" button (in the EDA's HistogramFilter) might be a good compromise?

In EDA's HistogramVisualization we could quite easily hook up the currently inoperative range-selection to set the x-axis range, if we want to.
